src/weathergen/utils/config.py

```
@@ -7,98 +7,162 @@
-            fname = Path(run_id)
-            logger.info(f"Loading config from provided full run_id path: {fname}")
-        else:
-            path_models = Path(model_path)
-            epoch_str = ""
-            if epoch is not None:
-                epoch_str = "_latest" if epoch == -1 else f"_epoch{epoch:05d}"
-            fname = path_models / run_id / f"model_{run_id}{epoch_str}.json"
+        fname = path_models / run_id / f"model_{run_id}{epoch_str}.json"
+
+    _logger.info(f"Loading config from specified run_id and epoch: {fname}")
 
-            logger.info(f"Loading config from specified run_id and epoch: {fname}")
+    with fname.open() as f:
+        json_str = f.read()
+        print(json_str)
 
-        # open the file and read into a config object
-        with fname.open() as f:
-            json_str = f.readlines()
+    return OmegaConf.create(json.loads(json_str))
 
-        cf = Config()
-        cf.__dict__ = json.loads(json_str[0])
 
-        return cf
+def load_config(
+    private_home: Path | None = None,
+    run_id: str | None = None,
+    epoch: int | None = None,
+    overwrite_path: Path | None = None,
+) -> Config:
+    private_config = load_private_conf(private_home)
+    overwrite_config = load_overwrite_conf(overwrite_path)
 
+    if run_id is None:
+        base_config = load_default_conf()
+        base_config.run_id = get_run_id()
+    else:
+        base_config = load_model_config(run_id, epoch, private_config["model_path"])
 
-def load_overwrite_conf(pth: str) -> dict:
+    # use OmegaConf.unsafe_merge if too slow
+    return OmegaConf.merge(base_config, private_config, overwrite_config)
+
+
+def load_overwrite_conf(overwrite_path: Path | None = None) -> Config:
     "Return the overwrite configuration."
+
     "If path is None, return an empty dictionary."
-    if not pth:
+    if overwrite_path is None or not overwrite_path.is_file():
```

I think a warning should be produced here, so it won't be a silent error

---

I would put a logging statement for both cases:
- loading a file / directory
- being passed none and indicating that no overwriting conf is loaded